types/network/index.d.ts

```diff
@@ -176,3 +185,135 @@ export type TagActionResponse = {
 	action: { op: string; id: string };
 	_jsns: string;
 };
+
+type NameOrIdRequired =
```

Comment:
The type defined as it is is not actually useful. The never rule is overwritten by the last definition, with both the fields required.

Also, it is convenient to create a type util to achieve this behaviour instead of replicating the conditions
```ts
type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
	{
		[K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
	}[Keys];
```
to be used as follow
```ts
type ModifyIdentityRequest = {
	_jsns: NameSpace.ZimbraAccount;
	identity: {
		_attrs?: Partial<IdentityAttrs>;
	} & RequireAtLeastOne<Pick<Identity, 'id' | 'name'>>;
};
```